### PR TITLE
GIF: select frame upon import

### LINKED ISSUE #98: Animated gif: select frame on decoding
Saving an animated gif to another file type will result in only the first frame being saved.

### PR DESCRIPTION
**Pull request by foresterre**
**Overview:**
If an image is an animated image, that is, the image consist of multiple frames which can be played in some order, we currently always drop all frames except for the first. The image buffer will thus contain a still image. 

Eventually we would like to provide support for animated images. 
This PR is the first step to support animated images. It deals with selecting a custom frame which will be extracted to our (still) image buffer.  

A user can select a frame by using the `--select-frame <FRAME>` argument.
`<FRAME>` allowed values:
* one-indexed unsigned integer  (`1..usize::MAX`): to select a frame by the index
* `first`: to select the first frame
* `last`: to select the last frame

Importing (loading) images was rewritten to accommodate for this change. The implementation has been simplified, and specifically now uses the `Read` trait instead of using indirections like `ExportMethod`. The images are now also always first read into a buffer and then imported as image from the buffer, instead of:
* if we have a path: open from the path (by using the extension to decide on the format) OR
* if we have no path: import from memory (by using headers to guess the format)
with the limitation that if an image had no extension (or an incorrect one), we would reject the input image.   
Of course, always guessing the format by its header may also result in similar limitations.

**In short**:
* This PR: allow to extract a custom frame from an animated image (GIF only; as the image crate doesn't support APNG or other animated formats).
* A future PR: provide real support for animated images

---

closes #98

### CODE REVIEW

File: sic_io/src/load.rs [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+use std::io::{BufReader, Read};
+use std::path::Path;
+
+use sic_core::image;
+use sic_core::image::AnimationDecoder;
+
+/// Load an image using a reader.
+/// All images are currently loaded from memory.
+pub fn load_image<R: Read>(
+    reader: &mut R,
```
Comment (foresterre) [inline sic_io/src/load.rs]: Just came across a newly introduced Reader for the image crate: https://github.com/image-rs/image/commit/ac93e755b51def00c93e7fadd33a5e5593afa1e2#diff-c42f25b7f2392d73d32de4183d660a9aR34

It is similar to the current implementation, so perhaps we can replace our implementation and use the Reader from the image crate on its next release.
